### ISSUE TITLE
PriorityQueue doesn't work with references

### ISSUE THREAD

**Issue by adam-arold** [korlibs/kds#18]
I tried to use the `PriorityQueue` in this lib, but it seems that it only works with numbers and `TGen` objects. I have no idea what are those. I cannot supply a `Comarable` type, or a `Comparator` either.

**Comment by soywiz** [comment]
`TGen` is just a generic so you can use it on any type as long it is comparable or you provide a comparator.
https://github.com/korlibs/kds/blob/9a806c089a809ae661c5483d47cd92617cd4f9dd/kds/src/commonMain/kotlin/com/soywiz/kds/PriorityQueue.kt#L9-L10

You have samples on how to use it in tests here:
https://github.com/korlibs/kds/blob/9a806c089a809ae661c5483d47cd92617cd4f9dd/kds/src/commonTest/kotlin/com/soywiz/kds/PriorityQueueTest.kt

**Comment by adam-arold** [comment]
I'm not following. Where do I import this `TGen` from? My classes are not inheritors of `TGen`. It should be a generic type parameter, like in Java: `PriorityQueue<T>` or `PriorityQueue<T : Comparable>`. Did I miss something?

**Comment by soywiz** [comment]
`T` is just the name of the generic, instead of `PriorityQueue<T>` is `PriorityQueue<TGen>`. It could be `T`, `K`, `L`, `A` , `TGen` or whatever. The typical convention is just `T` but nothing prevents you from using whatever you want.
I used TGen if I remember correctly to make it easier to replace `TGen` , `<TGen>` and other stuff to specialize stuff at the source generation level.

But again, check the link I provided. There you have samples on how the use the class.
